pkg/sql/create_index.go

```diff
@@ -585,6 +590,15 @@ func maybeCreateAndAddShardCol(
 	if err != nil && !columnIsUndefined {
 		return nil, err
 	}
+
+	if !cv.IsActive(clusterversion.Start22_1) {
+		return nil, pgerror.Newf(
+			pgcode.FeatureNotSupported,
+			"cannot create hash sharded index on a node running binary newer than 21.2 in a mixed version cluster "+
+				"during a cluster upgrade from pre-22.1. Please run the schema change on a node still running pre-22.1 binary "+
+				"or wait until the cluster upgrade finalized to 22.1. Hash sharded indexes have been improved in 22.1 to use "+
+				"a virtual column and can be added more efficiently")
```

Comment:
Use `errors.WithHint` and `errors.WithDetail` to break down this message into smaller parts, this is too long.